### PR TITLE
52 document api radiative transfer

### LINKED ISSUE #52: Document API: radiative transfer
Document all of the functions in the src/radiator directory

### PR DESCRIPTION
**Pull request by K20shores**
Link to [this branch's docs with my changes](https://ncar.github.io/photo-decomp/branch/52-document-api-radiative-transfer/api/radiative_transfer/).

This should close #52. Notably missing in this PR is the description of the configuration for grids, but that is because #48 seems to be a catch-all for configuration. I will certainly add the configuration information if it is desired here, but I thought this was specifically for the code documentation.

### CODE REVIEW

File: src/radiator.F90 [hunk 1]
```diff
@@ -1,11 +1,8 @@
 ! Copyright (C) 2020 National Center for Atmospheric Research
 ! SPDX-License-Identifier: Apache-2.0
-!
-!> \file
-!> The base_tuvx_radiator module
 
-!> The radiator_t type and related functions
 module tuvx_radiator
+! Represents a layer of radiation
```
Comment (mattldawson) [inline src/radiator.F90:5]:
```suggestion
! Represents an atmospheric constituent that affects radiative transfer calculations by absorbing or scattering radiation
```
@stacywalters - does this sound right?

File: src/radiator_warehouse.F90 [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 ! SPDX-License-Identifier: Apache-2.0
-!
-!> \file
-!> The tuvx_radiator warehouse module
 
-!> The radiator_warehouse_t type and related functions
 module tuvx_radiator_warehouse
+! A class that holds and gives out 
+! :f:type:`~tuvx_radiator/radiator_t`s built by the
+! :f:mod:`~tuvx_radiator_factory`.
```
Comment (mattldawson) [inline src/radiator_warehouse.F90:7]: for some reason this isn't being formatted correctly:
image
maybe it needs a space after the closing `?

File: src/radiative_transfer.F90 [hunk 3]
```diff
@@ -47,30 +43,28 @@ module tuvx_radiative_transfer
...
 
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 
-  !> radXfer component core constructor
-  !!
-  !! Sets up radXfer solver objects
-  function constructor( config, gridWareHouse, ProfileWareHouse ) result( radXfer_component )
+  function constructor( config, gridWareHouse, ProfileWareHouse ) & 
+    result( radXfer_component )
+    ! Initializes the componenents necessary for an n-stream
```
Comment (mattldawson) [inline src/radiative_transfer.F90:57]:
```suggestion
    ! Initializes the components necessary to solve radiative transfer
```

File: src/radiators/aerosol.F90 [hunk 4]
```diff
@@ -1,11 +1,8 @@
 ! Copyright (C) 2020 National Center for Atmospheric Research
 ! SPDX-License-Identifier: Apache-2.0
-!
-!> \file
-!> The aerosol_tuvx_radiator module
 
-!> The radiator_aerosol_t type and related functions
 module tuvx_radiator_aerosol
+! A radiator that calculates values based on aerosols
```
Comment (mattldawson) [inline src/radiators/aerosol.F90:5]:
```suggestion
! A radiator that accounts for the effects of aerosols on the radiation field
```

**Review by mattldawson — APPROVED** [review]
looks good! just a couple minor comments

File: src/radiator_warehouse.F90 (hunk 2 above)
Comment (K20shores) [reply src/radiator_warehouse.F90:7]: Done, had to add an apostrophe, which may not be grammatically correct but the space looked weird

**Comment by K20shores** [comment]
> this branch's docs with my changes

**Pull request merged**